Microsoft.HealthVault/Connection/HealthVaultConnectionBase.cs

```diff
@@ -178,6 +178,11 @@ protected HealthVaultConnectionBase(IServiceLocator serviceLocator)
             this.SessionCredential = await sessionCredentialClient.GetSessionCredentialAsync(token).ConfigureAwait(false);
         }
 
+        Task IConnectionInternal.RefreshSessionCredentialAsync(CancellationToken token)
+        {
+            return this.RefreshSessionCredentialAsync(token);
```

Comment:
No, there's additional logic that needs to be inside the lock in the other method that's calling it. Though I did just notice that we are sitting inside the lock when making the call retry, which we probably don't need to do.

---
In reply to: [115856865](https://github.com/Microsoft/healthvault-dotnetstandard-sdk/pull/27#discussion_r115856865) [](ancestors = 115856865,115836683,115805026,115799063)